test/common/TestSingleGradient.hs

```diff
@@ -423,8 +423,8 @@ foo (x,y,z) =
 
 testFoo :: Assertion
 testFoo =
-  assertEqualUpToEps (1e-10 :: Double)
-    (rev foo (1.1, 2.2, 3.3))
+  assertEqualUpToEpsilon (1e-10 :: Double)
+    (rev foo (1.1, 2.2, 3.3) :: (Double,Double,Double))
```

Comment:
It would be great to get rid of the extra type annotation. Perhaps define the class for `Double` and `Float` instead of for `a`? Perhaps then you can get rid of `OVERLAPPABLE` as well?

Reply:
Hm I did just that, but it doesn't help. :/

Reply:
And what's the error message?

Reply:
```
test/common/TestSingleGradient.hs:428:6: error:
    • Couldn't match type: Dual
                             'ADModeGradient
                             (mono-traversable-1.0.15.3:Data.MonoTraversable.Element a1)
                     with: HordeAd.Internal.Delta.Delta0
                             (mono-traversable-1.0.15.3:Data.MonoTraversable.Element a1)
        arising from a use of ‘rev’
      The type variable ‘a1’ is ambiguous
    • In the second argument of ‘assertEqualUpToEpsilon’, namely
        ‘(rev foo (1.1, 2.2, 3.3))’
      In the expression:
        assertEqualUpToEpsilon
          (1e-10 :: Double) (rev foo (1.1, 2.2, 3.3))
          (2.4396285219055063, - 1.953374825727421, 0.9654825811012627)
      In an equation for ‘testFoo’:
          testFoo
            = assertEqualUpToEpsilon
                (1e-10 :: Double) (rev foo (1.1, 2.2, 3.3))
                (2.4396285219055063, - 1.953374825727421, 0.9654825811012627)
    |
428 |     (rev foo (1.1, 2.2, 3.3))
    |
```

Reply:
Thank you. That looks hard. After you merge, I will try to remember how I use the `MonoTraversable` and experiment to see if I can help the type family `Dual` resolve.